src/traffic_cache_tool/CacheScan.cc

```diff
@@ -378,20 +378,31 @@ CacheScan::get_alternates(const char *buf, int length)
       } else if (!a->m_request_hdr.m_http) {
         std::cerr << "no http object found in the request header object" << std::endl;
         return zret;
-      } else if (((char *)a->m_request_hdr.m_http - buf) > length) {
+      } else if (!doc_mem.contains((char *)a->m_request_hdr.m_http)) {
         std::cerr << "out of bounds request header in the alternate" << std::endl;
         return zret;
       }
 
       auto *url = a->m_request_hdr.m_http->u.req.m_url_impl;
       if (check_url(doc_mem, url)) {
         std::string str;
-        ts::bwprint(str, "stripe: {} : {}://{}:{}/{};{}?{}", std::string_view(this->stripe->hashText),
-                    std::string_view(url->m_ptr_scheme, url->m_len_scheme), std::string_view(url->m_ptr_host, url->m_len_host),
-                    std::string_view(url->m_ptr_port, url->m_len_port), std::string_view(url->m_ptr_path, url->m_len_path),
-                    std::string_view(url->m_ptr_params, url->m_len_params), std::string_view(url->m_ptr_query, url->m_len_query));
 
-        std::cout << str << std::endl;
+        if (search) {
+          ts::bwprint(str, "{}://{}:{}/{};{}?{}", std::string_view(url->m_ptr_scheme, url->m_len_scheme),
```

Comment:
You should be able to unify these two cases better. You can do this formatting in both cases. For the non-search case you can then do `std::cout << "stripe: " << this->stripe->hashText << str << std::endl;`

Reply:
I did the concatenation to avoid the std out getting mixed up.